junit-platform-runner/src/main/java/org/junit/platform/runner/JUnitPlatform.java

```diff
@@ -255,6 +266,19 @@ private void addExcludedEnginesFilter(LauncherDiscoveryRequestBuilder requestBui
 		return patterns;
 	}
 
+	private String[] getExcludeClassNamePatterns() {
+		String[] patterns = getValueFromAnnotation(ExcludeClassNamePatterns.class, ExcludeClassNamePatterns::value,
+			new String[0]);
+
+		if (patterns.length == 0) {
+			return new String[] {};
```

Comment:
No need to create a new array here. That was done in the previous line. So just return `patterns` if its length is `0`.